test/test_sparse.py

```diff
@@ -30,60 +30,28 @@
 # batched grad doesn't support sparse
 gradcheck = functools.partial(gradcheck, check_batched_grad=False)
 
-def cpu_only(inner):
-    @functools.wraps(inner)
-    def outer(self, *args, **kwargs):
-        if self.is_cuda:
-            raise unittest.SkipTest("Test is CPU-only")
-        inner(self, *args, **kwargs)
-    return outer
-
-
-def cuda_only(inner):
-    @functools.wraps(inner)
-    def outer(self, *args, **kwargs):
-        if not self.is_cuda:
-            raise unittest.SkipTest("Test is GPU-only")
-        inner(self, *args, **kwargs)
-    return outer
-
 class TestSparse(TestCase):
 
     def setUp(self):
-        # These parameters control the various ways we can run the test.
-        # We will subclass and override this method to implement CUDA
-        # tests
-        self.is_cuda = False
-        self.is_uncoalesced = False
-        self.device = 'cpu'
-        self.exact_dtype = True
-        self.value_dtype = torch.float64
-        self.index_tensor = lambda *args: torch.tensor(*args, dtype=torch.int64, device=self.device)
-        self.value_empty = lambda *args: torch.empty(*args, dtype=self.value_dtype, device=self.device)
-        self.value_tensor = lambda *args: torch.tensor(*args, dtype=self.value_dtype, device=self.device)
+        self.index_tensor = lambda *args, **kwargs: torch.tensor(*args, **kwargs, dtype=torch.int64)
```

Comment:
This is an interesting style to assign a function to an attribute of this class.

Why isn't this a helper function?

Reply:
It could be a helper function, previously `self.index_tensor` was just an alias for int64 tensors for specific `self.device` value. Now is the same except that  device value can change.